src/search/landmarks/landmark_cost_partitioning_heuristic.cc

```diff
@@ -0,0 +1,159 @@
+#include "landmark_cost_partitioning_heuristic.h"
+
+#include "landmark_cost_assignment.h"
+#include "landmark_factory.h"
+#include "landmark_status_manager.h"
+
+#include "../plugins/plugin.h"
+#include "../task_utils/successor_generator.h"
+#include "../task_utils/task_properties.h"
+#include "../utils/markup.h"
+
+#include <cmath>
+#include <limits>
+
+using namespace std;
+using utils::ExitCode;
+
+namespace landmarks {
+LandmarkCostPartitioningHeuristic::LandmarkCostPartitioningHeuristic(
+    const plugins::Options &opts)
+    : LandmarkHeuristic(opts) {
+    if (log.is_at_least_normal()) {
+        log << "Initializing landmark cost partitioning heuristic..." << endl;
+    }
+
+    /*
+      TODO: The landmark graph is already constructed by the base class for
+       *LandmarkHeuristic*. The only reason why we get the landmark factory from
+       the options again is to check whether it produces reasonable orderings.
+       This could also be done by checking the landmark graph itself whether it
+       has any reasonable orderings at all.
+
+       In any case, doing this check only here means the landmark graph was
+       already computed by the base class, which would not be necessary if we
+       could ask the factory first, whether it computes reasonable orderings in
+       the first place. Since this all happens in the constructor anyway and we
+       abort if reasonable orderings are present, I don't think this is a big
+       issue, though.
+    */
```

Comment:
I would shorten this comment: remove the second paragraph and the last sentence of the first, and instead add a different sentence:
TODO: The landmark graph is already constructed by the base class for
       \*LandmarkHeuristic\*. The only reason why we get the landmark factory from
       the options again is to check whether it produces reasonable orderings.
       The check for reasonable orders can be removed once landmark progression 
      correctly handles them (issue1036).

I don't know how long computing the landmark graph takes, but I can imagine it can take some time for large tasks, so it can be annoying to get this error message only after a while even though we could check it much faster. We could instead add a function "check_options_consistency" or similar to the base class (either abstract or returns true), which is called right at the beginning of the base class constructor. Then every derived class can define custom consistency checks. (In this case the comment becomes obsolete)

Reply:
The constructor of the base class complained (legitimately) that it cannot call a virtual function before constructing the object. We discussed this offline and decided to add arguments to the constructor of the base class that tell it whether reasonable orderings, axioms, and conditional effects are supported by the concrete heuristic.